### ISSUE TITLE
too many terms in compound SELECT, when relating over 500 items.

### ISSUE THREAD

**Issue by laurenskling** [strapi/strapi#20402]
### Required System information

Environment: development
OS: darwin-arm64
Strapi Version: 4.20.3
Node/Yarn Version: yarn/1.22.19 npm/? node/v18.20.2 darwin arm64
Edition: Community
Database: sqlite

### Describe the bug

When linking a lot relations, sqlite throws error: `too many terms in compound SELECT`
according to https://stackoverflow.com/questions/9527851/sqlite-error-too-many-terms-in-compound-select you can only use 500 unions. Im trying to link 572 relations at one, thus crashing.

### Steps to reproduce the behavior

1. Have a model with a relation
2. add over 500 relations (from code is probably easier than from the UI ;))
3. See error:
```SqliteError: insert into `vehicle_b_2_b_vehicle_b_2_b_mail_users_links` (`user_id`, `user_order`, `vehicle_b_2_b_mail_id`) select 1 as `user_id`, 0 as `user_order`, 28 as `vehicle_b_2_b_mail_id` union all select 4 as `user_id`, 1 as `user_order`, 28 as `vehicle_b_2_b_mail_id` union all select 7 as `user_id`, 2 as `user_order`, 28 as `vehicle_b_2_b_mail_id` union all select 8 as `user_id`, 3 as `user_order`, 28 as `vehicle_b_2_b_mail_id` union all select 16 as `user_id`, 4 as `user_order`, 28 as  ....- too many terms in compound SELECT```

### Expected behavior

I'd like to add as many relations as I want :)

### related issues:

- https://github.com/strapi/strapi/issues/14157

**Comment by Marc-Roig** [comment]
I believe this issue should only happen with sqlite, is that the case?

**Comment by laurenskling** [comment]
Yes this is because sqlite doesn't accept over 500 items in a compound. Postgres accepts more, but I don't know if that is unlimited. 
Batching would relieve this issue. Just as we did in https://github.com/strapi/strapi/issues/22318

**Comment by TheRakeshPurohit** [comment]
By when to expect a patch for this?

**Comment by laurenskling** [comment]
> > By when to expect a patch for this?
> 
> You can try this:
> 
> https://docs.strapi.io/cms/api/document-service/fields#update
> 
> Just add any field on select. Works for me

this has absolutely nothing to do with the issue presented

**Comment by instatdigital** [comment]
> > > By when to expect a patch for this?
> > 
> > 
> > You can try this:
> > https://docs.strapi.io/cms/api/document-service/fields#update
> > Just add any field on select. Works for me
> 
> this has absolutely nothing to do with the issue presented

you not presented any example, i tried to give some example, when it helps. Just present your case. You use connect? Or Set?

**Comment by instatdigital** [comment]
in presented error you have too much union select. If you add strapi fields select it will decrease amount of selects

**Comment by laurenskling** [comment]
That's not the select that is failing. The select is the relations. Please stop guesstimating. The problem is clear. The solution is batching.

**Comment by minht11** [comment]
This issue is blocking our upgrade as well. We have 20k items. Tried to manually lower batchLimit to 100 like in https://github.com/strapi/strapi/issues/22318 but that did not help.

**Comment by TheRakeshPurohit** [comment]
I migrated from SQLite to MySQL without assets with strapi export feature
 - so I took backup
 - exported the data
 - moved connection
 - imported data in minutes
 - Moved assets manually with same path `/public`
 - Upgraded from `patch` -> `minor` -> `major`
 - Remove v4 plugin that are no longer in v5

**Comment by laurenskling** [comment]
Because in v5 every publish now updates all relations, this issue makes updating an entry with over 500 relations [crash](https://github.com/strapi/strapi/issues/25198)

**Comment by laurenskling** [comment]
Fixed with https://github.com/strapi/strapi/pull/25390
